text/0011-utxo-model/0011-utxo-model.md

```diff
@@ -0,0 +1,194 @@
+  
++ **Scalable Layer1 Smart Contracts:** Balances can be extended by a sequence of dynamic unlock conditions that will
+enable the use of a non-turing complete scripting language for smart contracts on layer1, which would not just be run
+by a small committee of validators but by everybody. This will enable things like:
+
+    + decentralized gambling
+    
+    + decentralized exchanges
+    
+    + hashed time lock contracts (enabling i.e. "lightning network"-like scaling)
+    
+    + regulatory compliant tokenized assets
+    
+    + ...
+
+# Detailed design
+
+Instead of keeping track of aggregated balances per address, the **Unspent Transaction (TX) Output** model stores
+balances associated to **outputs** that are the result of individual transactions moving funds and that can be addressed
+and spent individually. This allows to exactly specify which funds are getting moved even if multiple parties are
+funding an address in parallel.
+
+A transaction moving funds consists out of three building blocks:
+
+- the ``Inputs`` which defines a list of ``OutputIDs`` that reference the *consumed* ``Outputs`` from previous
+  transactions (or the genesis).
+- the ``Outputs`` which defines where and how many of the consumed tokens are moved.
+- the ``Unlock Section`` that contains data used to *unlock* the consumed inputs and *authorize* spends (usually just
+the address signatures - more on that later).
+
+The following diagram shows how transactions consume outputs as inputs to create new outputs for future transactions:
+
+![sdf](images/utxo_flow_of_funds.png)
+
+### Output IDs
+
+The transaction ID that creates the output together with some _additional identifier_ that distinguishes different
+outputs from the same transaction uniquely identifies every output.
+
+Bitcoin (who introduced the UTXO model) uses the numerical index of the output inside a transaction as this
+*additional identifier*. It does not actually have a concept of balances belonging to addresses, and providing a
+signature for a certain address is simply a special type of *unlocking condition* for this particular output. This is a
+very generic way of keeping track of balances that is to a certain degree even agnostic to the use of addresses as the
+main way of interacting with such a system.
+
+Even though Bitcoin has this kind of inherent agnosticism towards using addresses, it has proven to be extremely
+beneficial to at least have some form of *logical identifier* that groups related outputs together and that can also be
+used to *authorize* payments in the form of address signatures. It is for that very reason that the predominant part of
+transaction executed on today's DLTs make use of addresses.
+
+To build a bridge between the worlds of *UTXO-* and *account-based* ledgers, we define our UTXO variant to use
+addresses instead of output indexes as this additional identifier:
+
+``OutputID = Pair<Address, TransactionID>``
```

Comment:
Do you want to have 2 indexes for the output to be used internally by the node when it tracks utxos?
Or do you want this to be the actual id to be used inside the transaction that we create?

I understood that you want the latter. If so, I just don't understand the motivation. I think it just bloats up the transaction. If I understood correctly, then could you please elaborate on why it is useful?